vispy/visuals/volume.py

```diff
@@ -585,32 +673,25 @@ def clim(self, value):
         if not (clim.ndim == 1 and clim.size == 2):
             raise ValueError('clim must be a 2-element array-like')
         self._clim = tuple(clim)
-        if self.texture_is_inverted:
-            if (clim[0] > self._texture_limits[0]) or (clim[1] < self._texture_limits[1]):
-                self.rescale_data()
-                return
-        else:
-            if (clim[0] < self._texture_limits[0]) or (clim[1] > self._texture_limits[1]):
-                self.rescale_data()
-                return
-        # if the clim range is too small of a percentage of the last-used texture range,
-        # posterization may be visible, so downscale the texture range.
-        range_ratio = np.subtract(*clim) / abs(np.subtract(*self._texture_limits))
-        if np.abs(range_ratio) < self._clim_range_threshold:
+        if self._check_clim_requires_rescale(clim):
```

Comment:
Comment for above but we should accept a tuple. An array is a bit overkill.
`try: cmin, cmax = value except: ...`

Reply:
Yeah this code does seem weird. Convert to array just to check the shape then back to a tuple on the instance attribute. I'll have to walk through this code to make sure I'm not missing something with the array usage.

If I'm understanding you correctly, you'd like things refactored to just use cmin/cmax rather than a tuple or an array. And if we wanted to use a tuple, then don't accept an array at all. Right?

Reply:
Yes. But it's mostly cosmetic, not really that important as long as the code is correct.